frb_rust/src/pool.rs

```diff
@@ -0,0 +1,227 @@
+//! Original: wasm_bindgen raytrace-parallel example
+//!
+//! File: https://github.com/rustwasm/wasm-bindgen/blob/main/examples/raytrace-parallel/src/pool.rs
```

Comment:
nit: Given that we have also copy-pasted some other code in other files, shall we create a new folder, say, `wasm_bindgen_copied`, and put all copy-and-paste code into there. By doing this, it is very easy to see what we have copied and manage them